packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core-plugin-api';
+import { IconComponent } from '../../icons';
+
+/**
+ * API for accessing app icons.
+ *
+ * @public
+ */
+export interface IconsApi {
+  getIcon(key: string): IconComponent | undefined;
+
+  listIconKeys(): { keys: Iterable<string> };
```

Comment:
Iterable isn't particularly convenient. Would be worth just making an array, I'd bet.

Also, while we do like object form results, the method name is VERY specific, so may just consider making it string[]

Reply:
Yeah it's not intended to be called particularly often after all.

Happy to make it a simple return, another option I considered was more of a `getIconsMeta(): { ... }`. Didn't wanna get stuck for too long though

Reply:
Switched to `(): string[]`